scripts/idl/generators/cpp/BridgeClustersCpp.jinja

```diff
@@ -0,0 +1,240 @@
+#include <app-common/zap-generated/cluster-id.h>
+#include <app-common/zap-generated/attribute-id.h>
+
+#include <new>
+
+namespace clusters {
+{% for struct in structs %}
+struct {{struct.name}}
+{
+  CHIP_ERROR Decode(chip::TLV::TLVReader & reader)
+  {
+    chip::app::Clusters::detail::Structs::{{struct.name}}::DecodableType t;
+    CHIP_ERROR err = t.Decode(reader);
+    if(err == CHIP_NO_ERROR) {
+      {%-      for field in struct.fields %}
+      {{field.name}} = t.{{field.name}};
+      {%-      endfor  %}
+    }
+    return err;
+  }
+
+  CHIP_ERROR Encode(chip::TLV::TLVWriter & writer, chip::TLV::Tag tag) const
+  {
+    chip::app::Clusters::detail::Structs::{{struct.name}}::Type t;
+    {%-      for field in struct.fields %}
+    t.{{field.name}} = {{field.name}};
+    {%-      endfor  %}
+    return t.Encode(writer, tag);
+  }
+
+  {%-   for field in struct.fields %}
+  {{field | getField(None, idl)}} {{field.name}};
+  {%-  endfor  %}
+};
+{%- endfor %}
+
+{%- for cluster in clusters %}
```

Comment:
Can we split files into per-cluster or similar? 

ZAP "generate single file" is very painful in compile times and debuggability/inclusion. Could we generate one cluster struct (and potentially encode/decode) for each cluster? I would rather do things like `#include <LightBridge>` than `#include <AllBridges>`

Reply:
There's still a single include point the generated cluster & attribute list does the including but it's split out now